### ISSUE TITLE
Phase 1: Neural Noise — Replace FBM with Trained MLP in Terrain Generation

### ISSUE THREAD

**Issue by plagness** [plagness/Metalcraft#11]
## Summary

Replace the 4 hand-tuned `fbm2D` passes in `VoxelTerrainGen.metal` with a small trained MLP (Multi-Layer Perceptron) that generates richer, more realistic terrain height components.

## Background

Current terrain height computation in `terrainHeight()` (VoxelTerrainGen.metal:245-259) uses:
- `continental = fbm2D(pos * 0.004, seed, 4 octaves)` — low-frequency continent shapes
- `hills = fbm2D(pos * 0.008, seed+300, 4 octaves)` — medium features
- `detail = fbm2D(pos * 0.04, seed+400, 3 octaves)` — fine detail
- `ridged = abs(fbm2D(pos * 0.0032, seed+500, 3 octaves))` — ridge mountains

This produces smooth but monotonous terrain. A trained MLP can output the same 4 noise channels but with learned patterns from real-world heightmap data (NASA SRTM), enabling: river valleys, erosion patterns, cliff faces, layered mesas, volcanic craters.

## Network Architecture

```
Input:  4 float — worldX, worldZ (normalized), seed_hash, biome_dominant
Hidden: 2 × 32 neurons, ReLU activation
Output: 4 float — continental, hills, detail, ridged
Parameters: 4×32 + 32 + 32×32 + 32 + 32×4 + 4 = 1,316 weights (~2.6 KB FP16)
```

### Metal Shader Changes (`VoxelTerrainGen.metal`)
- [ ] Add MLP inference function `neuralNoiseLayers()` using matrix multiply + ReLU
- [ ] Load weights from constant buffer `[[buffer(2)]]`
- [ ] Replace 4 `fbm2D` calls in `terrainHeight()` with `neuralNoiseLayers()`
- [ ] Keep legacy `fbm2D` path behind a toggle for A/B comparison

### Swift Integration (`GPUTerrainGenerator.swift`)
- [ ] Add `MTLBuffer` for neural weights, loaded from `neural_noise_weights.bin`
- [ ] Bind weights buffer at index 2 in `generate()` and `generateBatch()`
- [ ] Add `useNeuralNoise: Bool` flag to `TerrainConfig`

### Training Pipeline (`Tools/TrainNoiseNet/`)
- [ ] Python script to generate training data from NASA SRTM heightmaps
- [ ] PyTorch MLP training script
- [ ] Export script: PyTorch weights → flat FP16 binary

## Performance Target

| Metric | Current (FBM) | Target (Neural) |
|--------|--------------|-----------------|
| Ops per column | ~320 MAD | ~1,316 MAD |
| Terrain variety | Low (smooth FBM) | High (learned features) |
| Determinism | ✅ | ✅ (same weights = same output) |
| Memory | 0 | ~2.6 KB (weights buffer) |

## Verification
- Screenshot comparison: same seed, FBM vs Neural
- GPU profiling: terrain generation time per chunk
- Determinism test: generate same chunk twice, compare block-by-block
- Build and run successfully on M1 (baseline)

## Files to Modify
- `VoxelEngine/Shaders/Voxel/VoxelTerrainGen.metal`
- `VoxelEngine/Compute/GPUTerrainGenerator.swift`
- New: `Tools/TrainNoiseNet/train.py`
- New: `Resources/neural_noise_weights.bin`
